src/View/mainpage/DicomView.py

```diff
@@ -91,13 +91,18 @@ def image_display(self):
         label = QtWidgets.QGraphicsPixmapItem(image)
         self.scene = GraphicsScene(label, self.horizontal_view, self.vertical_view)
 
-    def draw_roi_polygons(self, roi_id, polygons):
+    def draw_roi_polygons(self, roi_id, polygons, roi_color=None):
```

Comment:
can you do roi_color=self.roi_color? so that we dont need to have the logic in line 102-105

Reply:
ah, I did this because the newly created ROI has had a color stored in the self.roi_color dictionary yet. That's why I added the third parameter to set a temporary color for the new ROI. Furthermore, if I set roi_color for the dicom_view to avoid the hassle in lines 102-105, all currently selected ROIs in the MainWindow will automatically appear in the ROI preview window in the ManipulateROIWindow. So the answer is no unless you have another idea.